### PR TITLE
Fix --version flag issue for create managed nodegroup command

### LINKED ISSUE #3806: `--version` flag for creating managed nodegroups with specific k8s version is not setting k8s version
## Managed nodegroup inherits cluster's k8s version despite `--version` being set
Creating a managed nodegroup with a k8s version that is different from the cluster k8s version is not really doing anything: the nodegroup inherits the version of the cluster.

Output of trying to create a managed ng with a k8s version that is different from that of the cluster: 
```
eksctl create nodegroup --cluster test-2 --name test-ng-2 --managed --version 1.19
2021-06-04 12:32:59 [ℹ]  eksctl version 0.52.0
2021-06-04 12:32:59 [ℹ]  using region us-west-2
2021-06-04 12:33:00 [!]  will use version 1.19 for new nodegroup(s), while control plane version is 1.18; to automatically inherit the version use "--version=auto"
2021-06-04 12:33:09 [ℹ]  1 existing nodegroup(s) (test-ng) will be excluded
2021-06-04 12:33:09 [ℹ]  1 nodegroup (test-ng-2) was included (based on the include/exclude rules)
2021-06-04 12:33:09 [ℹ]  will create a CloudFormation stack for each of 1 managed nodegroups in cluster "test-2"
2021-06-04 12:33:09 [ℹ]  2 sequential tasks: { fix cluster compatibility, 1 task: { 1 task: { create managed nodegroup "test-ng-2" } } }
2021-06-04 12:33:09 [ℹ]  checking cluster stack for missing resources
2021-06-04 12:33:10 [ℹ]  cluster stack has all required resources
2021-06-04 12:33:10 [ℹ]  building managed nodegroup stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:33:10 [ℹ]  deploying stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:33:10 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:33:27 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:33:44 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:34:05 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:34:23 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:34:44 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:35:04 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:35:23 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:35:42 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:36:00 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:36:18 [ℹ]  waiting for CloudFormation stack "eksctl-test-2-nodegroup-test-ng-2"
2021-06-04 12:36:19 [ℹ]  no tasks
2021-06-04 12:36:19 [✔]  created 0 nodegroup(s) in cluster "test-2"
2021-06-04 12:36:20 [ℹ]  nodegroup "test-ng-2" has 2 node(s)
2021-06-04 12:36:20 [ℹ]  node "ip-192-168-52-123.us-west-2.compute.internal" is ready
2021-06-04 12:36:20 [ℹ]  node "ip-192-168-78-122.us-west-2.compute.internal" is ready
2021-06-04 12:36:20 [ℹ]  waiting for at least 2 node(s) to become ready in "test-ng-2"
2021-06-04 12:36:20 [ℹ]  nodegroup "test-ng-2" has 2 node(s)
2021-06-04 12:36:20 [ℹ]  node "ip-192-168-52-123.us-west-2.compute.internal" is ready
2021-06-04 12:36:20 [ℹ]  node "ip-192-168-78-122.us-west-2.compute.internal" is ready
2021-06-04 12:36:20 [✔]  created 1 managed nodegroup(s) in cluster "test-2"
2021-06-04 12:36:22 [ℹ]  checking security group configuration for all nodegroups
2021-06-04 12:36:22 [ℹ]  all nodegroups have up-to-date configuration
```

In the output above, there is a log that kind of acknowledges that the nodegroup being created has a different version to that of the cluster, but in the end the nodegroup still inherits the cluster's k8s version.

Another issue is that, currently, the k8s version of a nodegroup is not being returned when we do `get nodegroup`. This means that the only way to find the actual k8s version of a nodegroup is via the console (edit: #3810 adds capability for returning the nodegroup's k8s version!)

### PR DESCRIPTION
**Pull request by vchetu**
Fixes #3806 

### Description

--version flag is not supported for creating managed nodegroup as k8s version is inherited from control plane.
This flag is valid only for unmanaged nodegroups.

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by nikimanoledaki** [comment]
Thanks for opening this PR, @vchetu, this is great and also closes https://github.com/weaveworks/eksctl/issues/3806! 

Could you also please update the user docs [here](https://github.com/weaveworks/eksctl/blob/main/userdocs/src/usage/managing-nodegroups.md#managing-nodegroups) where it has a note about `--version`? :)

**Comment by vchetu** [comment]
> Thanks for opening this PR, @vchetu, this is great!
> Could you also please update the user docs [here](https://github.com/weaveworks/eksctl/blob/main/userdocs/src/usage/managing-nodegroups.md#managing-nodegroups) where it has a note about `--version`? :)

@nikimanoledaki Updated the docs. Thank you

**Comment by Callisto13** [comment]
Thanks @vchetu, could you add a test in `pkg/ctl/create/nodegroup_test.go`?

**Comment by vchetu** [comment]
> Thanks @vchetu, could you add a test in `pkg/ctl/create/nodegroup_test.go`?

@Callisto13 Added a unit test. Thank you

File: userdocs/src/usage/managing-nodegroups.md [hunk 1]
```diff
@@ -7,9 +7,11 @@ To create an additional nodegroup, use:
 ```
 eksctl create nodegroup --cluster=<clusterName> [--name=<nodegroupName>]
 ```
-
+ 
 !!!note
-    By default, new nodegroups inherit the version from the control plane (`--version=auto`), but you can specify a different
+    `--version` flag is not supported for managed nodegroup. It always inherits the version from control plane.
```
Comment (cPu1) [inline userdocs/src/usage/managing-nodegroups.md:12]: Minor nit:
```suggestion
    `--version` flag is not supported for managed nodegroups. It always inherits the version from control plane.
```

**Review by cPu1 — APPROVED** [review]
I have left a minor comment but otherwise LGTM.

File: userdocs/src/usage/managing-nodegroups.md (hunk 1 above)
Comment (vchetu) [reply userdocs/src/usage/managing-nodegroups.md]: @cPu1 Updated the user document. Thank You

**Review by cPu1 — APPROVED** [review]

**Pull request merged**